src/Cube/Heuristics/StateOne.hs

```diff
@@ -1,9 +1,83 @@
-module Cube.Heuristics.StateOne
-( binaryFold
-) where
+module Cube.Heuristics.StateOne where
+
+import Cube.Edges
+import Cube.Moves ((-:))
+import Cube.Moves.Edges
+import qualified Data.HashMap.Strict as M
+import qualified Data.Vector.Unboxed as V
+
+{-
+ - A list of all 18 possible moves under the half-turn metric (HTM)
+ -}
+moves :: [Edge -> Edge]
+moves = [
+      front
+    , back
+    , left
+    , right
+    , up
+    , down
+    , front . front
+    , back . back
+    , left . left
+    , right . right
+    , up . up
+    , down . down
+    , front . front . front
+    , back . back . back
+    , left . left . left
+    , right . right . right
+    , up . up . up
+    , down . down . down
+    ]
 
 {-
  - This is used as a reducer for the edge orientation vector.
  - It treats the vector as a binary number and converts it to decimal using little-endian.
  -}
 binaryFold carry index value = carry + value * 2 ^ index
+
+{-
+ - Static lookup list. This is to be generated from a script ... TODO
+ -}
+heuristicList :: M.HashMap Int Int
+heuristicList = M.singleton 0 0
+
+{-
+ - Generate a list of edge states by applying all moves to the provided state
+ -}
+nextStates :: Edge -> [Edge]
+nextStates edge = map (edge -:) moves
+-- if any of the states are all zeros, discard them since theres nothing to come from it
+{- nextStates edge = filter (\a -> V.sum (orien a) /= 0) $ map (\f -> f edge) moves -}
+
+{-
+ - Map the provided list of edge states to their corresponding heuristic index value
+ - TODO: this differs from the paper because the vector is reversed
+ -}
+heuristicIndices :: [Edge] -> [Int]
+heuristicIndices = map $ V.ifoldl binaryFold 0 . V.init . orien
```

Comment:
I feel like the usage of this function and the one above could be combined in a more elegant fashion.... TODO